client/tink_keyset_helper.go

```diff
@@ -42,3 +46,35 @@ func nameOfSupportedTinkKeyTemplates() string {
 	sort.Strings(supportedTemplates)
 	return strings.Join(supportedTemplates, "\n")
 }
+
+// checkTemplateNameAndKnoxIDForTinkKeyset checks whether knox identifier start with "tink:<tink_primitive_short_name>:".
+func checkTemplateNameAndKnoxIDForTinkKeyset(templateName string, knoxIentifier string) error {
+	templateInfo, existed := tinkKeyTemplates[templateName]
+	if !existed {
+		return errors.New("not supported keyset template. See 'knox key-templates'")
+	} else if !strings.HasPrefix(knoxIentifier, templateInfo.knoxIDPrefix) {
+		return errors.New("<key_identifier> must have prefix '" + templateInfo.knoxIDPrefix + "'")
```

Comment:
from a UX perspective-- do we want users to specify the prefixes or should we just automatically add them?

Reply:
I feel we should require users to explicitly add the prefix to the key_identifier. I can see users getting confused if we auto add those prefixes. It's a one time thing per key so shouldn't be a lot of work for the users.

Reply:
Hi Kalpesh, thank you for your comment!

Hi Jeremy, I agree with Kalpesh, if we add the prefix automatically when the user-provided Knox ID is illegal, user may feel confused.